server/route.go

```diff
@@ -491,6 +481,20 @@ func routeSid(sub *subscription) string {
 	return fmt.Sprintf("%s%s:%d:%s", qi, RSID, sub.client.cid, sub.sid)
 }
 
+func parseRouteSid(rsid []byte) (cid uint64, sid []byte, ok bool) {
+	if !bytes.HasPrefix(rsid, []byte(QRSID_PREFIX)) {
+		return
+	}
+
+	for i := QRSID_PREFIX_LEN; i < len(rsid); i++ {
+		switch rsid[i] {
+		case ':':
+			return uint64(parseInt64(rsid[QRSID_PREFIX_LEN:i])), rsid[i+1:], true
```

Comment:
If the server would receive a corrupted message, like `QRSID:1:`, the server would crash. You need to check that there is actually something past the `:` character. I believe that the previous use of the regex would have return false and so server would not crash.